pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java

```diff
@@ -137,25 +165,35 @@ public SchedulerManager(WorkerConfig workerConfig, PulsarClient pulsarClient, Pu
     }
 
     public Future<?> schedule() {
-        return executorService.submit(() -> {
-            synchronized (SchedulerManager.this) {
-                boolean isLeader = membershipManager.isLeader();
-                if (isLeader) {
-                    try {
-                        invokeScheduler();
-                    } catch (Exception e) {
-                        log.warn("Failed to invoke scheduler", e);
-                        throw e;
+        try {
```

Comment:
I think we need to simplify this massively.
I think part of the pr that I'm working on wrt metadata simplification will impact this as well.

Reply:
What are you thinking?  What is the complexity here?